src/praisonai-agents/tests/unit/test_jira_tools.py

```
@@ -0,0 +1,287 @@
+
+
+class TestJIRAConnection:
+    """Test JIRA connection utility."""
+    
+    @patch('praisonaiagents.tools.jira_tools.JIRA')
+    def test_connection_with_email_token(self, mock_jira):
+        """Test JIRA connection with email and token."""
+        _get_jira_connection(
+            url="https://test.atlassian.net",
+            email="test@example.com", 
+            token="test_token"
```

Ruff (0.15.15)

[error] Possible hardcoded password assigned to argument: "token"

(S106)